src/Web/Pages.cpp

```diff
@@ -332,6 +353,36 @@ void Pages::error(AsyncWebServerRequest* request)
  * Local Functions
  *****************************************************************************/
 
+/**
+ * Set the factory partition active to be considered as the boot partition.
+ *
+ * @return True if factory partition was set active successfully, otherwise false.
+ */
+static bool setFactoryPartitionActive()
+{
+    bool                   isSuccessful = false;
+    const esp_partition_t* partition    = esp_partition_find_first(
+        esp_partition_type_t::ESP_PARTITION_TYPE_APP,
+        esp_partition_subtype_t::ESP_PARTITION_SUBTYPE_APP_FACTORY,
+        nullptr);
+
+    if (nullptr != partition)
+    {
+        esp_err_t err = esp_ota_set_boot_partition(partition);
+
+        LOG_INFO("Setting factory partition '%s' as boot partition", partition->label);
+        isSuccessful = true;

```

Comment:
The sequence here is a litte odd.  You indicate success, then check result and eventually signal error in isSuccessfull. Restructure to only set it true when all actions pass.  Readability is more important then source code compactness.